src/app/reporting/Engine.h

```diff
@@ -102,6 +104,17 @@ class Engine
      */
     static void Run(System::Layer * aSystemLayer, void * apAppState);
 
+#if !CHIP_SYSTEM_CONFIG_NO_LOCKING
+    class ScopedLock
+    {
+    public:
+        ScopedLock(Engine & aEngine) : mEngine(aEngine) { mEngine.mAccessLock.Lock(); }
```

Comment:
scoped locks should be on mutexes not other types

Reply:
removed, thanks

Reply:
removed, thanks